code/modules/projectiles/guns/projectile.dm

```diff
@@ -195,28 +197,31 @@
 		to_chat(user, "<span class='warning'>[src] is empty.</span>")
 	update_icon()
 
-/obj/item/gun/projectile/attackby(var/obj/item/A as obj, mob/user as mob)
+/obj/item/gun/projectile/attackby(obj/item/A, mob/user)
 	..()
 	load_ammo(A, user)
 
-/obj/item/gun/projectile/attack_self(mob/user as mob)
-	if(is_jammed)
-		to_chat(user, "<span class='notice'>\The [user] unjams \the [src]!</span>")
-		if(do_after(user, 5))
-			playsound(src.loc, 'sound/weapons/empty.ogg', 100, 1)
-			is_jammed = 0
+/obj/item/gun/projectile/attack_self(mob/user)
+	if(jam_num)
+		playsound(src.loc, 'sound/weapons/empty.ogg', 50, 1)
+		jam_num--
+		if(!jam_num)
+			visible_message(span("danger", "\The [user] unjams \the [src]!"))
+			playsound(src.loc, 'sound/items/glowstick.ogg', 100, 1)
+			unjam_cooldown = world.time
+	else if(unjam_cooldown + 2 SECONDS > world.time)
```

Comment:
This might stop the gun from chaging modules or unloading ammo. Should just check if the gun is jammed at all before checking all of this stuff.